### PR TITLE
Proposed solution for #155 [BUG] iOS camera preview crashes when switching app

### PR DESCRIPTION
**Pull request by lcuis**
This proposed solution solved my issue on iPhone SE 2020 through near-production installation through test-flight.

### CODE REVIEW

**Comment by lcuis** [comment]
Here is what I did to validate this on my iPhone SE 2020:
1. uninstall example app
2. run example app with the correction
3. open photo example
4. validate permission
=> camera preview displayed as expected

I did the same with uninstall for the tap video example. I got 2 an additional permission request for the microphone and the camera preview also displayed as expected.

Is there another scenario you had in mind?

**Comment by AlexV525** [comment]
> Is there another scenario you had in mind?

Yes, more cases like when previewing the captured picture then throwing the app back to the background and opening it again could cause related issues.

**Comment by lcuis** [comment]
I just tried the following with success on my iPhone SE 2020:
1. run plugin example app
2. take a picture (first example)
3. while staying on the taken image preview, switch to other app
4. switch back to example app
=> As expected, taken image preview is still displayed. Also, when going back, the camera preview is displayed as expected.

File: lib/src/states/camera_picker_state.dart [hunk 1]
```diff
@@ -199,12 +199,16 @@ class CameraPickerState extends State<CameraPicker>
   @override
   void didChangeAppLifecycleState(AppLifecycleState state) {
     final CameraController? c = innerController;
+    if (c == null && state == AppLifecycleState.resumed) {
```
Comment (AlexV525) [inline lib/src/states/camera_picker_state.dart:202]: Can we merge the null condition with the below one?

Comment (lcuis) [reply lib/src/states/camera_picker_state.dart:202]: This should work:
```dart
    if (c == null || !c.value.isInitialized) {
        if (state == AppLifecycleState.resumed) {
            initCameras(currentCamera);
        }
        else {
            // App state changed before we got the chance to initialize.
            return;
        }
    }
```

Let me try it now on my device.

Comment (lcuis) [reply lib/src/states/camera_picker_state.dart]: It worked and I updated the PR accordingly.

**Comment by AlexV525** [comment]
Please verify if the latest change still works fine.

**Comment by lcuis** [comment]
I confirm that the formatted version works as well for the mentioned scenarios on my iPhone SE 2020.

**Pull request merged**